components/tenant-fetcher/cmd/main.go

```diff
@@ -65,29 +83,67 @@ func main() {
 		exitOnError(errors.New("missing handler endpoint or tenant path parameter"), "Error while loading app handler config")
 	}
 
+	transact, closeFunc, err := persistence.Configure(ctx, cfg.Database)
+	exitOnError(err, "Error while establishing the connection to the database")
+
+	authenticatorsConfig, err := authenticator.InitFromEnv(envPrefix)
+	exitOnError(err, "Failed to retrieve authenticators config")
+
+	defer func() {
+		err := closeFunc()
+		exitOnError(err, "Error while closing the connection to the database")
+	}()
+
+	uidSvc := uuid.NewService()
+	converter := tenant.NewConverter()
+	repo := tenant.NewRepository(converter)
+	service := tenant.NewService(repo, transact, uidSvc)
+
 	ctx, err = log.Configure(ctx, &cfg.Log)
 	exitOnError(err, "Failed to configure Logger")
 	logger := log.C(ctx)
 
+	middleware := auth.New(
+		cfg.JwksEndpoint,
+		cfg.CISIdentityZone,
+		cfg.SubscriptionCallbackScope,
+		extractTrustedIssuersScopePrefixes(authenticatorsConfig),
+		cfg.AllowJWTSigningNone,
+	)
+
+	if cfg.JWKSSyncPeriod != 0 {
```

Comment:
Isn't this check redundant because the default JWK sync period is 5 minutes?

Reply:
Addressed it. 🚀